src/commands/Minion/farm.ts

```diff
@@ -0,0 +1,313 @@
+			altProtection: true,
+			oneAtTime: true,
+			cooldown: 1,
+			usage: '[quantity:int{1}|name:...string] [plantName:...string]',
+			usageDelim: ' '
+		});
+	}
+
+	@minionNotBusy
+	@requiresMinion
+	async run(msg: KlasaMessage, [quantity, plantName = '']: [null | number | string, string]) {
+		if (msg.flagArgs.plants) {
+			return msg.channel.sendFile(
+				Buffer.from(
+					Farming.Plants.map(
+						plant =>
+							`${plant.seedType}: ${plant.name} -- lvl ${
+								plant.level
+							}: ${Object.entries(plant.inputItems)
+								.map(entry => `${entry[1]} ${itemNameFromID(parseInt(entry[0]))}`)
+								.join(', ')}\n	Default # of patches: ${
+								plant.defaultNumOfPatches
+							}\n${plant.additionalPatchesByFarmLvl.map(
+								entry =>
+									`	| Farming Level: ${entry[0]} => Total Additional Patches: ${entry[1]} |\n`
+							)} ${plant.additionalPatchesByQP.map(
+								entry =>
+									`	| Quest Points: ${entry[0]} => Total Additional Patches: ${entry[1]} |\n`
+							)} `
+					).join('\n')
+				),
+				`Farming Plants.txt`
+			);
+		}
+
+		await msg.author.settings.sync(true);
+		const userBank = msg.author.settings.get(UserSettings.Bank);
+		const questPoints = msg.author.settings.get(UserSettings.QP);
+		const GP = msg.author.settings.get(UserSettings.GP);
+		const currentWoodcuttingLevel = msg.author.skillLevel(SkillsEnum.Woodcutting);
+		const currentDate = new Date().getTime();
+
+		let payment = false;
+		let upgradeType = '';
+		let str = '';
+		let upgradeStr = '';
+		let paymentStr = '';
+		let boostStr = '';
+		const timePerPatch = Time.Minute * 1.5;
+
+		if (typeof quantity === 'string') {
+			plantName = quantity;
+			quantity = null;
+		}
+
+		const plants = Farming.Plants.find(plants =>
+			plants.aliases.some(
+				alias =>
+					stringMatches(alias, plantName) || stringMatches(alias.split(' ')[0], plantName)
+			)
+		);
+
+		if (!plants) {
+			throw `That's not a valid seed to plant. Valid seeds are ${Farming.Plants.map(
+				plants => plants.name
+			).join(', ')}. *Make sure you are not attempting to farm 0 crops.*`;
+		}
+
+		if (msg.flagArgs.supercompost) {
+			upgradeType = 'supercompost';
+			upgradeStr += `You are treating all of your patches with supercompost. `;
+		}
+		if (msg.flagArgs.ultracompost) {
+			upgradeType = 'ultracompost';
+			upgradeStr += `You are treating all of your patches with ultracompost. `;
+		}
+		if (msg.flagArgs.pay) {
+			payment = true;
+			paymentStr += `You are paying a nearby farmer to look after your patches. `;
+		}
+
+		if (msg.author.skillLevel(SkillsEnum.Farming) < plants.level) {
+			throw `${msg.author.minionName} needs ${plants.level} Farming to plant ${plants.name}.`;
+		}
+
+		if (plants.canPayFarmer === false && payment === true) {
+			throw `You cannot pay a farmer to look after your ${plants.name}s!`;
+		}
+		if (
+			plants.canCompostandPay === false &&
+			payment === true &&
+			upgradeType === ('supercompost' || 'ultracompost')
```

Comment:
You dont need to compare to `true` or `false`, and the third condition doesnt do what you think it does.